### ISSUE TITLE
free transformer

### ISSUE THREAD

**Issue by lucidrains** [lucidrains/x-transformers#345]
note for self

- [x] change encoder to masked averaging for variable length sequences
- [x] add what seems to be a learned query attention pool in their diagram for encoder / figure out if it is or can be per token
- [x] projected latent condition needs to be summed to the first kv branch. unsure how important this is - also add ability to sum to pre-softmax attention, for a new [contrastive attention decoding paper in iclr 2026](https://openreview.net/forum?id=uomCTwGflg)
- [x] make sure kv caching works at inference
- [x] move binary mapper to vq-pytorch for reuse - also redo so aux loss always returned
